app/models/pipeline_run.rb

```diff
@@ -1577,6 +1578,22 @@ def precache_report_info!
     end
   end
 
+  def precache_report_info_v2!
+    params = report_info_params
+    Background.top_for_sample(sample).pluck(:id).each do |background_id|
+      cache_key = PipelineReportService.report_info_cache_key(
+        "/samples/#{sample.id}/report_v2",
+        params.merge(background_id: background_id)
+      )
+      Rails.logger.info("Precaching #{cache_key} with background #{background_id}")
+      Rails.cache.fetch(cache_key, expires_in: 30.days) do
+        PipelineReportService.call(self, background_id)
+      end
+
+      MetricUtil.log_analytics_event("samples.cache.precached", current_user)
```

Comment:
also not sure who `current_user` would be in this example, can probably leave it blank

Reply:
`log_analytics_event` requires a user to be passed in, so I replaced it with `nil`.